docs/backend/guides/how-to/create-a-custom-linter.md

```diff
@@ -0,0 +1,137 @@
+# How to Create a Custom Go Linter
+
+Creating custom GO linters can be a great way to analyze your 
+project's source code and alert you to bugs, errors, 
+or other issues with your code.
+
+## Setting up your linter
+
+### File structure
+Start by setting up the files you'll need for your linter:
+```golang
+-cmd
+    -linter_name
+        -main.go
+
+-pkg
+    -linter_name
+        -linter.go
+        -linter_test.go
+```
+In the `cmd` folder create a folder for your linter called `<linter_name>` and add an empty file called `main.go`
+
+In `pkg` folder create another folder for your linter called `<linter_name>` and in it you will 
+have your `linter.go` and `linter_test.go`
+
+### Analyzer:
+In the `cmd` folder contains the the analysis driver for your linter. The file `main.go` will utilize a package called `singlechecker`,
+which defines the main function for an analysis driver and provides a tool to run a standalone analysis.
+
+The code in `main.go` is very simple:
+
+```golang
+package main
+
+import (
+	"example.org/linter_name"
+	"golang.org/x/tools/go/analysis/singlechecker"
+)
+
+func main() { singlechecker.Main(linter_name.LinterAnalyzer) }
+```
+
+### Creating a linter
+The linter will live in the `pkg` folder, in a folder named after your linter. 
+In `linter.go` you will store the linter analyzer that gets referenced in `cmd/linter_name/main.go`. 
+It will contain the name of the liner, documentation about the linter, a call to runthe linter, and requirements:
+
+```golang
+var LinterAnalyzer = &analysis.Analyzer{
+	Name:     "linternamelint",
+	Doc:      "Make sure X object is properly used throughout codebase",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+}
+```
+
+Note that `Run` expects an interface and runs your linter code. This value can be the function that executes your linter. In this example, we are calling teh function `run`:
+
+```golang
+func run(pass *analysis.Pass) (interface{}, error) {
+    // Some code
+}
+```
+
+The linter is actually analyzing an abstract syntax tree, AST, that represents code in a file. 
+When your linter gets to a position in a file, where it catches an error, bug, or issue, it will flag this for the user.
+Because the linter is analyzing an AST, your code must be able to search through a file and mark which position it is when an issue is caught.
+
+A great resource to use to learn how to write code for analyzing an AST can be found [here](https://disaev.me/p/writing-useful-go-analysis-linter/).
+If you're curious about what an AST looks like for your program, checkout [this](http://goast.yuroyoro.net/) online visualizer.
```

Comment:
It's generally good to include more descriptive text in links for accessibility. See this [A11y Project post on accessible links](https://www.a11yproject.com/posts/2019-02-15-creating-valid-and-accessible-links/).

Reply:
It might also be good to state that there may be slight differences between what the online AST visualizer shows and what the see when working on their linter. But it's still helpful to start getting used to what it looks like.